File: tests/ulcer_index_test.py
import numpy as np

from taflow import UlcerIndex


def test_lifecycle() -> None:
    values = np.linspace(10.0, 1.0, 32)
    state = UlcerIndex(5).extend(values)
    assert len(state) == len(values)
    assert state.reset().value is None

File: crates/taflow-core/src/stream/ulcer_index_test.rs
use super::ulcer_index::UlcerIndex;

#[test]
fn warmup_and_reset_are_consistent() {
    let mut state = UlcerIndex::new(4).unwrap();
    for value in [10.0, 9.0, 8.0, 9.0, 7.0] {
        state.append(value);
    }
    assert!(state.value().is_some());
    state.reset();
    assert_eq!(state.value(), None);
}
